falcon/request.py

```diff
@@ -571,6 +568,35 @@ def get_header(self, name, required=False):
 
             raise HTTPMissingParam(name)
 
+    def get_header_as_datetime(self, header, required=False):
+        """Return an HTTP header with HTTP-Date values as a datetime.
+
+        Args:
+            name (str): Header name, case-insensitive (e.g., 'Date')
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning gracefully when the
+                header is not found (default ``False``).
+
+        Returns:
+            datetime: The value of the specified header if it exists,
+                or ``None`` if the header is not found and is not required.
+
+        Raises:
+            HTTPBadRequest: The header was not found in the request, but
+                it was required.
+            HttpInvalidHeader: The header contained a malformed/invalid value.
+        """
+
+        try:
+            http_date = self.get_header(header, required=required)
+            return util.http_date_to_dt(http_date)
+        except TypeError:
```

Comment:
Catching an exception will be more expensive (in terms of CPU time) than explicitly checking whether `http_date is None`. If we expect the header to almost always exist in the request, the current model is ideal because it avoids the (usually) superfluous check against None. If the header will often not be found, relying on exceptions can be less performant.

That being said, much of the overhead of exception handling is incurred in the setup (when you wrap in a "try" block), and since we need to do that anyway to catch ValueError, I think the code should be fine as-is. :+1: